pkg/dataplane/item.go

```diff
@@ -80,3 +83,57 @@ func (i Item) GetFieldUint64(name string) (uint64, error) {
 		return 0, v3ioerrors.ErrInvalidTypeConversion
 	}
 }
+
+// For internal use only - DO NOT USE!
+func (i Item) GetShard() (int, []*ItemChunkData, *ItemChunkMetadata, *ItemCurrentChunkMetadata, error) {
+	streamDataPrefix := "__data_stream["
+	streamMetadataPrefix := "__data_stream_metadata["
+
+	var chunkDataArray []*ItemChunkData
+	chunkMetaData := ItemChunkMetadata{}
+	currentChunkMetadata := ItemCurrentChunkMetadata{}
+	var chunkID int
+
+	for k, v := range i {
+		if strings.HasPrefix(k, streamDataPrefix) {
+			chunkID, _ = strconv.Atoi(k[len(streamDataPrefix):][:4])
```

Comment:
i think you are missing a case where multiple chunks arrive on the same response..

Reply:
the for loop does this iteration over all the chunks in the same response